commit fd8ca9bff298f9d17b585c7cf58d78031d92b99f
Author: Peter Michael Green <plugwash@debian.org>
Date:   Sun Nov 20 06:54:18 2022 +0000

    Fix memory reporting with sysinfo 0.16
    
    sysinfo 0.16 changed the units for reporting memory from kilobytes to bytes,
    this commit updates the reported units to match. I also added TB to the list
    of possible units. Computers with terabytes of memory do exist nowadays.

diff --git a/src/feature/si.rs b/src/feature/si.rs
index 94ff40a..aecd2e9 100644
--- a/src/feature/si.rs
+++ b/src/feature/si.rs
@@ -301,9 +301,11 @@ fn check_user(_process: &Process, _user: &User) -> bool {
 #[cfg(feature = "si")]
 fn suffix(val: usize) -> &'static str {
     match val {
-        0 => "KB",
-        1 => "MB",
-        2 => "GB",
+        0 => "B",
+        1 => "KB",
+        2 => "MB",
+        3 => "GB",
+        4 => "TB",
         _ => "xB",
     }
 }
@@ -354,10 +356,12 @@ mod test {
 
     #[test]
     fn suffix_works() {
-        assert_eq!("KB", suffix(0));
-        assert_eq!("MB", suffix(1));
-        assert_eq!("GB", suffix(2));
-        assert_eq!("xB", suffix(3));
+        assert_eq!("B", suffix(0));
+        assert_eq!("KB", suffix(1));
+        assert_eq!("MB", suffix(2));
+        assert_eq!("GB", suffix(3));
+        assert_eq!("TB", suffix(4));
+        assert_eq!("xB", suffix(5));
     }
 
     #[test]
